enterprise/frontend/src/metabase-enterprise/moderation/components/CreateModerationIssuePanel.jsx

```diff
@@ -1,46 +1,74 @@
 import React, { useState } from "react";
+import { connect } from "react-redux";
 import PropTypes from "prop-types";
+import cx from "classnames";
 
 import { useAsyncFunction } from "metabase/lib/hooks";
 import { MODERATION_TEXT } from "metabase-enterprise/moderation/constants";
 import {
   getModerationStatusIcon,
   getColor,
+  getUserTypeTextKey,
 } from "metabase-enterprise/moderation";
+import { getIsModerator } from "metabase-enterprise/moderation/selectors";
+
 import Icon from "metabase/components/Icon";
 import Button from "metabase/components/Button";
 
 CreateModerationIssuePanel.propTypes = {
   issueType: PropTypes.string.isRequired,
   onReturn: PropTypes.func.isRequired,
   createModerationReview: PropTypes.func.isRequired,
+  createModerationRequest: PropTypes.func.isRequired,
   itemId: PropTypes.number.isRequired,
-  itemType: PropTypes.string.isRequired,
+  isModerator: PropTypes.bool.isRequired,
+  moderationRequest: PropTypes.object,
 };
 
 function CreateModerationIssuePanel({
   issueType,
   onReturn,
   createModerationReview: _createModerationReview,
+  createModerationRequest: _createModerationRequest,
   itemId,
-  itemType,
+  isModerator,
+  moderationRequest,
 }) {
   const [description, setDescription] = useState("");
   const icon = getModerationStatusIcon(issueType);
   const color = getColor(issueType);
+  const textColorClass = `text-${color}`;
+  const userType = getUserTypeTextKey(isModerator);
 
-  const [createModerationReview, isPending] = useAsyncFunction(
+  const [createModerationReview, isModerationReviewPending] = useAsyncFunction(
     _createModerationReview,
   );
+  const [
+    createModerationRequest,
+    isModerationRequestPending,
+  ] = useAsyncFunction(_createModerationRequest);
+
+  const isPending = isModerationRequestPending || isModerationReviewPending;
 
   const onCreateModerationReview = async e => {
     e.preventDefault();
 
-    await createModerationReview({
-      status: issueType,
-      moderated_item_id: itemId,
-      moderated_item_type: itemType,
-    });
+    if (isModerator) {
+      await createModerationReview({
+        moderationReview: {
+          type: issueType,
+          cardId: itemId,
+          description,
+        },
+        moderationRequest,
+      });
+    } else {
+      await createModerationRequest({
+        type: issueType,
+        cardId: itemId,
+        description,
+      });
+    }
```

Comment:
NIP: the function is called `onCreateModerationReview`, but if a user is not a moderator, the moderation _request_ is going to be created. I guess it can be a bit confusing. Still, the component and the function are really small and I don't think it will be a problem

Reply:
true, renaming the parent fn was a miss on my part. I'll update the name to `createModerationIssue` -- "issue" is the word I'm using to refer to either a review or a request. thanks for catching that